src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2513,6 +2515,360 @@ public Expand(Func<TIn, IEnumerator<TOut>> extrapolate)
         public override string ToString() => "Expand";
     }
 
+    public sealed class NotYetThereSentinel : Exception
+    {
+        public static readonly NotYetThereSentinel Instance = new();
```

Comment:
Realistically, we should use a single sentinel exception in general, it is more polite for when a stage using the concept may have ValueType `TIn`/`TOut`/`TMat`.